openbao: init at 2.0.3

brianmay opened the pull request:
Add OpenBao as a package with the initial version as 2.0.1.

Homepage: https://openbao.org/
GitHub: https://github.com/openbao/openbao
Changelog: https://github.com/openbao/openbao/blob/main/CHANGELOG.md

This is has updates from <https://github.com/NixOS/nixpkgs/pull/334485>. Will close this PR if/when changes get merged there.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

brianmay commented:
I am not comfortable with the `postInstall = ` stuff. Not convinced any of this is required. Might remove it.

brianmay commented:
I added back the rename binary to `baob`. Not sure if this is the best way or not.

I think the version test should be working now.

brianmay commented:
@cipherboy Can I please get you to review this PR?

brianmay commented:
> Duplicate of https://github.com/NixOS/nixpkgs/pull/334485, even though slightly newer

I said as much already.

But https://github.com/NixOS/nixpkgs/pull/334485, seems stale, and it doesn't look like it is going to get updated any time soon. As I can't update that PR, I created a new PR.

brianmay commented:
Darwin openbao.passthru.tests process appears to be stuck. All other CI OK.

brianmay commented:
The original PR was closed.

cipherboy reviewed (approved):
Apologies, it took me a while to get to this. I've not tested this as I don't have a Nix environment but nothing looks surprising here comparing it with the documentation.

I will note we just released v2.0.3 today to fix some security issues but otherwise I'm happy if that is updated in a subsequent PR.

Thank you for this! I'll add it to the website's documentation once this lands.

brianmay commented:
Hopefully will update to 2.0.3 tomorrow.

KiaraGrouwstra commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 354366`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>openbao</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review-december/1711/24

Ben9986 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 354366`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>openbao</li>
  </ul>
</details>

wolfgangwalther reviewed (dismissed):
Please rebase your branch and squash the commits, so that you end up with two commits:
- `maintainers: add brianmay`
- `openbao: init at 2.0.3`

cipherboy commented on pkgs/by-name/op/openbao/package.nix:
```diff
@@ -0,0 +1,60 @@
+    "-X github.com/openbao/openbao/version.fullVersion=${version}"
+  ];
+
+  postInstall = ''
+    mv $out/bin/openbao $out/bin/bao
+  '';
```
@wolfgangwalther @brianmay I think that's fine. As a FYI, that build script is only used in development via `make`, [the `goreleaser` script](https://github.com/openbao/openbao/blob/c3173a055a6ebcb5a2912daf4d58620eac2ba001/.goreleaser-template.yaml#L11) sets `project_name` to `bao` which eventually filters down to the binary name.

I will note `-o` isn't a ldflag, it is a compiler flag. Not sure if there's an option for that somewhere in NixOS.

brianmay replied:
```
evaluation warning: `buildFlags`/`buildFlagsArray` are deprecated and will be removed in the 24.11 release. Use the `ldflags` and/or `tags` attributes instead of `buildFlags`/`buildFlagsArray`
```

If I try to add it to `ldflags` instead then I get errors that it can't find `a.out`.

brianmay committed:
openbao: init at 2.1.0

Release notes - https://github.com/openbao/openbao/releases/tag/v2.1.0
Changelog - https://github.com/openbao/openbao/blob/main/CHANGELOG.md

brianmay committed:
maintainers: add brianmay

brianmay commented on maintainers/maintainer-list.nix:
```diff
@@ -3198,6 +3198,13 @@
+    email = "brian@linuxpenguins.xyz";
+    github = "brianmay";
+    githubId = 112729;
+    keys = [ { fingerprint = "D636 5126 A92D B560 C627  ACED 1784 577F 811F 6EAC"; } ];
```
I think I messed up in the rebasing. Now fixed.

(and hopefully enabled signing by default now)
